spec/jobs/get_usps_proofing_results_job_spec.rb

```diff
@@ -478,6 +478,31 @@
           end
         end
 
+        context 'when the profile is deleted before proofing' do
+          before do
+            pending_enrollment.update!(profile_id: nil)
+            stub_request_passed_proofing_results
+          end
+
+          it 'updates the enrollment and ignores the non-existent profile' do
+            freeze_time do
+              pending_enrollment.update(
+                status_check_attempted_at: Time.zone.now - 1.day,
+                status_updated_at: Time.zone.now - 2.days,
+                )
+
+              expect { job.perform(Time.zone.now) }.not_to(raise_exception)
+
+              pending_enrollment.reload
+              expect(pending_enrollment.status_updated_at).to eq(Time.zone.now)
+              expect(pending_enrollment.status_check_attempted_at).to eq(Time.zone.now)
+              expect(pending_enrollment.status_check_completed_at).to eq(Time.zone.now)
+              expect(pending_enrollment.status).to eq(InPersonEnrollment::STATUS_PASSED)
```

Comment:
I feel like we should stop USPS from processing an enrollment associated with a nil profile. But how likely is this to happen again and are there places earlier in the pipeline that we could enact effective interventions to stop this from happening again. Kinda just thinking out loud here.

Reply:
Yeah, this is where it turns into a whole mess that hurts my brain. 😇

It mostly makes sense to cancel the enrollment, _but_ it likely would have just led to further confusion for this user as they went to the Post Office and were told that the barcode was no good. Maybe we would want to pair that with an email to the user or something notifying them that, since their profile was deleted, their enrollment was canceled? (Or, warn them as they try to delete their profile?)

I think that should probably be a separate story, though.